### PR TITLE
feat: add Main, MainMenu, DrawerHeader and Sidebar components, tests

### LINKED ISSUE #6: Add components: Main, MainMenu, MenuItem, DrawerHeader, Sidebar

### PR DESCRIPTION
**Pull request by pyphilia**
closes #6

### CODE REVIEW

File: src/MenuItem/MenuItem.test.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      expect(icon).toHaveLength(1)
+      const chevron = menuItem.find(ChevronRightIcon)
+      expect(chevron).toHaveLength(1)
+      const text = menuItem.find(ListItemText)
+      expect(text).toHaveLength(1)
+      expect(text.prop('primary')).toEqual('text')
+    })
+  })
+
+  // describe('menu is defined with custom component', () => {
```
Comment (juancarlosfarah) [inline src/MenuItem/MenuItem.test.js:74]: Can we remove this?

File: src/Loader/__snapshots__/Loader.test.js.snap [hunk 2]
```diff
@@ -0,0 +1,27 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (juancarlosfarah) [inline src/Loader/__snapshots__/Loader.test.js.snap:1]: Do you need snap files for the tests?

File: src/DrawerHeader/DrawerHeader.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import ChevronRightIcon from '@material-ui/icons/ChevronRight'
+import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction'
+import { DEFAULT_DIRECTION, DRAWER_HEADER_HEIGHT } from '../constants'
+
+const styles = () => ({
+  drawerHeader: {
+    height: DRAWER_HEADER_HEIGHT
+  },
+  secondaryAction: {
+    right: '5px'
```
Comment (juancarlosfarah) [inline src/DrawerHeader/DrawerHeader.js:15]: Can we not use theme.spacing?

File: src/DrawerHeader/__snapshots__/DrawerHeader.test.js.snap [hunk 4]
```diff
@@ -0,0 +1,27 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (juancarlosfarah) [inline src/DrawerHeader/__snapshots__/DrawerHeader.test.js.snap:1]: Do we need snap files for the tests?

File: src/Header/Header.js [hunk 5]
```diff
@@ -40,15 +46,17 @@ const Header = ({ id, classes, handleDrawerOpen, isSidebarOpen }) => {
...
-          onClick={handleDrawerOpen}
-          className={clsx(classes.menuButton, isSidebarOpen && classes.hide)}
-        >
-          <MenuIcon />
-        </IconButton>
+        {hasSidebar && (
+          <IconButton
+            id={id}
+            color='inherit'
+            aria-label='Open drawer'
```
Comment (juancarlosfarah) [inline src/Header/Header.js:53]: Maybe use translation already? `t('Open Drawer')` not sure how feasible it is.

**Review by juancarlosfarah — CHANGES_REQUESTED** [review]
Very nice! Just some minor changes and a couple of questions.

File: src/Loader/__snapshots__/Loader.test.js.snap (hunk 2 above)
Comment (pyphilia) [reply src/Loader/__snapshots__/Loader.test.js.snap:1]: @juancarlosfarah I think it is the "only" way to test standalone components? That way we can easily test for default parameters and defined parameters. Let me know if you are thinking about something else

File: src/Header/Header.js (hunk 5 above)
Comment (pyphilia) [reply src/Header/Header.js:53]: I will change it with a prop where the default value will be `Open Drawer`.

File: src/Loader/__snapshots__/Loader.test.js.snap (hunk 2 above)
Comment (juancarlosfarah) [reply src/Loader/__snapshots__/Loader.test.js.snap:1]: @pyphilia, but you don't need the snap file of the test file itself. It's usually a snap file of the component file (e.g. `Loader.js.snap`). Or am I wrong?

File: example/src/App.tsx [hunk 6]
```diff
@@ -1,9 +1,16 @@
...
-import { Main } from '@graasp/ui'
+import { Main, MainMenu, MenuItem } from '@graasp/ui'
+import HomeIcon from '@material-ui/icons/Home'
 import '@graasp/ui/dist/index.css'
 
 const App = () => {
-  return <Main />
+  const mainMenu = (
+    <MainMenu>
+      <MenuItem text='Home' icon={<HomeIcon />} selected={true}></MenuItem>
```
Comment (juancarlosfarah) [inline example/src/App.tsx:9]: Factor out (and translate) "Home"?

File: src/texts.js [hunk 7]
```diff
@@ -0,0 +1 @@
+export const OPEN_DRAWER_LABEL = 'Open drawer'
```
Comment (juancarlosfarah) [inline src/texts.js:1]: `'Open Drawer'`. Do we need to translate this?

**Review by juancarlosfarah — CHANGES_REQUESTED** [review]
Great. Only very minor changes left.

File: example/src/App.tsx (hunk 6 above)
Comment (pyphilia) [reply example/src/App.tsx:9]: Do we really need to factor out and translate? This file is in `example`, but we can include translation if you think it is necessary (but for now the language cannot be changed in anyway).

File: src/texts.js (hunk 7 above)
Comment (pyphilia) [reply src/texts.js:1]: I'll change the capitalization. This text is used as default label for the drawer open button if no other text is submitted.

File: example/src/App.tsx (hunk 6 above)
Comment (juancarlosfarah) [reply example/src/App.tsx:9]: @pyphilia, no, that's OK. I missed that this was inside the example folder.

File: src/texts.js (hunk 7 above)
Comment (juancarlosfarah) [reply src/texts.js:1]: I think it's fine for now.

**Review by juancarlosfarah — APPROVED** [review]

**Pull request merged**